src/shogun/lib/SGStringList.cpp

```diff
@@ -93,12 +93,30 @@ SGStringList<T> SGStringList<T>::clone() const
 {
 	SGStringList<T> result(*this);
 
-	auto strings = SG_MALLOC(SGString<T>, num_strings);
+	auto strs = SG_MALLOC(SGString<T>, num_strings);
 
 	for (auto i : range(num_strings))
-		strings[i] = this->strings[i].clone();
+		strs[i] = this->strings[i].clone();
 
-	return SGStringList<T>(strings, num_strings, max_string_length);
+	return SGStringList<T>(strs, num_strings, max_string_length);
+}
+
+template <class T>
+bool SGStringList<T>::equals(const SGStringList<T>& other) const
+{
+	if (this->num_strings!=other.num_strings)
```

Comment:
maybe one thing to add would be if the pointer address and the number of strings is the same, then we can return true straight away

Reply:
exactly